sdr-enthusiasts/plane-alert-db
# ci: add GitHub action that automatically creates the derivative databases

This PR adds the '.github/workflows/create_db_derivatives.yaml'GitHub action that is used to automatically create the derivative databases (i.e. 'category', 'images' and 'extended') based on the 'plane-alert-db.csv', 'plane_images.txt' and 'blacklist.txt' files.

## Files explained

- **.github/workflows/create_db_derivatives.yaml**: A GitHub action file that runs every time somebody modifies the `plane-alert-db.csv`, `plane_images.txt` and `blacklist.txt` files.
- **scripts/create_db_derivatives.py**: A file that creates the derivative databases based on the `plane-alert-db.csv`, `plane_images.txt` and `blacklist.txt` files.
- **scripts/create_images_reference.py**: A small little script that I used to create the new `plane_images.txt` file. This file can be removed when we are sure that this file is correct.
- **plane_images:** This is the new `planepix.txt` file. I kept the file extension as `txt` to distinguish it from the main database and the derivative databases.
- **blacklist.txt:** Contains planes that had to be removed because of the new Twitter rules.
- **plane-alert-db-extended**: The `plane-alert-db.csv' database with the blacklist items added to it.
- **plane-alert-db-extended-images**: Same as above but now with the image links added.

## How to test

Before switching to this automatic workflow, we have to ensure that the generated databases are equal to the ones already in the repository. I compared them using [VScodes compare](https://vscode.one/diff-vscode/) feature, and the implementation looks correct to me. However, please double-check.

1. Go into the scripts folder.
2. Install the python dependencies.
3. Run the `create_images_reference.py` script to generate the database of the new image.
4. Run the `create_db_derivatives.py` script to generate the derivative databases.
5. Compare the generated databases.

## Things that could cause problems

My script currently replaces `N/a` with `np.nan`. I don't know how this field is read in the https://github.com/kx1t/docker-planefence repository, but otherwise, we have to add those values back in.

## Things that were not yet automated

 I did not know what was in the following databases:

- **badgers-best-images.csv**
- **badgers-best.csv**
- **plane-alert-wip.csv**
- **plane-alert-search-terms-to-do.csv**
- **plane-alert-wip.csv**
